codegen/module_system.go

```diff
@@ -134,6 +134,24 @@ func NewDefaultModuleSystem(
 		)
 	}
 
+	if err := system.RegisterClass("clients", module.Class{
```

Comment:
This is an interesting way to create a global clients object. You might want to pick a slightly more obvious variant of name here because client vs clients is pretty hard to distinguish at a glance. Nice to see that the SingleModule type actually still works because it isn't used anywhere right now.